content/sips/sip-307.md

```diff
@@ -67,15 +69,19 @@ Next, we describe other characteristics of the architecture.
 
 The main system proxy is an [UUPS proxy](https://eips.ethereum.org/EIPS/eip-1822) which is more gas efficient than transparent proxies by implementing the upgradeability functions in the implementation instead of the proxy. In this architecture, we develop a way of ensuring that all upgrades are further upgradeable via a special mechanism in the UpgradeModule.
 
-#### Satellites
+Since the upgradeability code in UUPS proxies is in the implemetation instead of the proxy, every update needs to make sure that the new system is upgradeable (unless of course the protocol wants to be intentionally frozen). Our implementation of UUPS proxies includes an on-chain mechanism to ensure that all new implementations are not sterile.
```

Comment:
I'm a little unclear if it will be possible to intentionally freeze upgrades on specific modules given the phrasing here.